src/adapter/src/coord/read_policy.rs

```diff
@@ -37,21 +38,24 @@ use timely::progress::Antichain;
 
 use crate::coord::id_bundle::CollectionIdBundle;
 use crate::coord::timeline::{TimelineContext, TimelineState};
-use crate::coord::Coordinator;
 use crate::session::Session;
 use crate::util::ResultExt;
 
 /// Relevant information for acquiring or releasing a bundle of read holds.
-#[derive(Debug, Serialize)]
+#[derive(Serialize)]
 pub struct ReadHolds<T> {
     pub holds: HashMap<Antichain<T>, CollectionIdBundle>,
+    // We only use Serialize for debugging.
+    #[serde(skip_serializing)]
+    dropped_read_holds_tx: Option<tokio::sync::mpsc::UnboundedSender<ReadHolds<T>>>,
```

Comment:
Is there a reason to send the whole `ReadHolds` struct instead of just sending a `HashMap<Antichain<T>, CollectionIdBundle>`? It feels a bit odd that we send a channel over the channel, but we have to remember that the sent channel is `None` (if I'm understanding that correctly).

Reply:
Yeah, sorry you had to see this. This is the bug I still have to fix. I'm sending a `ReadHolds` because that's what `release_read_holds(...)` takes. I had planned this approach for fixing it:

- Change `ReadHolds` back to what it was before.
- Add a new `DroppableReadHolds` that has a `ReadHolds` field and the channel, plus a `Deref` impl for accessing the underlying `ReadHolds`. That way I get rid of most of the cleaning-out-the-channel shenanigans.